internal/manifests/manifestutils/storage_test.go

```diff
@@ -330,3 +330,46 @@ func TestConfigureStorageWithS3CCO(t *testing.T) {
 	assert.NotContains(t, pod.Containers[0].Args, "--storage.trace.s3.secret_key=$(S3_SECRET_KEY)")
 	assert.NotContains(t, pod.Containers[0].Args, "--storage.trace.s3.access_key=$(S3_ACCESS_KEY)")
 }
+
+func TestAzureStorage_short_lived(t *testing.T) {
+	tempo := v1alpha1.TempoStack{
+		Spec: v1alpha1.TempoStackSpec{
+			Storage: v1alpha1.ObjectStorageSpec{
+				Secret: v1alpha1.ObjectStorageSecretSpec{
+					Name:           "test",
+					Type:           v1alpha1.ObjectStorageSecretAzure,
+					CredentialMode: v1alpha1.CredentialModeToken,
+				},
+			},
+		},
+	}
+
+	pod := corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name: "ingester",
+			},
+		},
+	}
+
+	params := StorageParams{
+		CredentialMode: tempo.Spec.Storage.Secret.CredentialMode,
+		AzureStorage: &AzureStorage{
+			TenantID:  "test-tenant",
+			ClientID:  "test-client",
+			Container: "tempo",
+		},
+	}
+	assert.NoError(t, ConfigureStorage(params, tempo, &pod, "ingester"))
+	assert.Len(t, pod.Containers[0].Env, 4)
```

Comment:
I think it would be cleaner to assert on the entire array of the env vars

Reply:
+1